### PR TITLE
BREAKING CHANGE: change return type of Deno.resources()

### PR DESCRIPTION
**Pull request by bartlomieju**
Ref #4882

### CODE REVIEW

File: cli/js/lib.deno.ns.d.ts [hunk 1]
```diff
@@ -2134,21 +2134,20 @@ declare namespace Deno {
...
+   *       console.log(Deno.resources());
+   *       // { 0: "stdin", 1: "stdout", 2: "stderr" }
    *       Deno.openSync('../test.file');
-   *       console.log(Deno.resources()); // e.g. { 0: "stdin", 1: "stdout", 2: "stderr", 3: "fsFile" }
+   *       console.log(Deno.resources());
+   *       // { 0: "stdin", 1: "stdout", 2: "stderr", 3: "fsFile" }
    */
-  export function resources(): ResourceMap;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  export function resources(): any;
```
Comment (bartlomieju) [inline cli/js/lib.deno.ns.d.ts:2150]: Maybe we should move this method to `Deno.internal`?

Comment (ry) [reply cli/js/lib.deno.ns.d.ts:2150]: I like it as a public api

File: cli/js/lib.deno.ns.d.ts [hunk 2]
```diff
@@ -2134,19 +2134,20 @@ declare namespace Deno {
    */
   export function metrics(): Metrics;
 
-  /** **UNSTABLE**: reconsider representation. */
   interface ResourceMap {
-    [rid: number]: string;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    [rid: number]: any;
```
Comment (ry) [inline cli/js/lib.deno.ns.d.ts:2139]: It would be useful to have a comment here explaining why we’re using any instead of string.

Comment (bartlomieju) [reply cli/js/lib.deno.ns.d.ts:2139]: PTAL at docstring for resources()

**Pull request merged**